tools/buildgen/extract_metadata_from_bazel_xml.py

```diff
@@ -1010,6 +1010,14 @@ def _detect_and_print_issues(build_yaml_like):
     bazel_rules.update(
         _extract_rules_from_bazel_xml(_bazel_query_xml_tree(query)))
 
+# TODO(veblush): Remove this workaround once upb is supported well
+# for both Bazel and non-Bazel (https://github.com/grpc/grpc/issues/24904)
+google_api_upbdefs_rule = bazel_rules["//:google_api_upbdefs"]
```

Comment:
Since in this hack you are manually adding a descriptor.upbdefs.c and descriptor.updefs.h depedency for a specific target `//:google_api_upbdefs`, it also means that you'd have to add   this hack for all the libraries that depend on `upb_lib_descriptor` and it seems that that's something that's hard to maintain and also something that over time people will run into (and they won't be able to solve this themselves as this whole issue is quite cryptic).

I don't know of a better solution right now (I'd have to spend some time thinking about it and experimenting), but at least you could:
- introduce a separate function (e.g. `_inject_ubp_descriptor_dependency`) with a proper description that explains why this hack is being used. I think this hack being applied at the top level is an invitation for folks to add more similar hacks (and I don't like that). Also the function should make it explicit that there are some libraries (currently it's just `//:google_api_upbdefs` but it might be more of them in the future) where the upb descriptor.proto dependency needs to be added manually. If there is a reason why only `:google_api_upbdefs` will ever need the upb_lib_descriptor dependency, please include that reasoning in the comment in said function.

Reply:
Idea: Maybe there's some way to adjust the logic from https://github.com/grpc/grpc/pull/24925/files 
to fix this without actually requiring to manually list the libraries that depend on descriptor.updefs.c and descriptor.upbdefs.h?

Reply:
It turns out that this is unnecessary because `src/upb/gen_build_yaml.py` already included it. Thanks!